packages/iTwinUI-react/src/core/ComboBox/ComboBox.tsx

```diff
@@ -180,18 +221,46 @@ export const ComboBox = <T,>(props: ComboBoxProps<T>) => {
     });
   }
 
+  // Get indices of selected elements in options array when we have selected values.
+  const getSelectedIndexes = () => {
+    const indexArray: number[] = [];
+    if (isMultipleEnabled(valueProp, multiple)) {
+      valueProp?.forEach((value) => {
+        indexArray.push(
+          optionsRef.current.findIndex((option) => option.value === value),
+        );
+      });
+    }
+    return indexArray;
+  };
+
   // Reducer where all the component-wide state is stored
   const [{ isOpen, selectedIndex, focusedIndex }, dispatch] = React.useReducer(
     comboBoxReducer,
     {
       isOpen: false,
-      selectedIndex: valueProp
+      // if multiple enabled -> get selected indexes, if not -> is valueProp defined -> yes - get index of selected value -> no - set index as -1
+      selectedIndex: isMultipleEnabled(valueProp, multiple)
```

Comment:
Maybe instead of this ternary with comment, we can move everything to `getSelectedIndexes` that would have normal if else.